Add trust token resource (#512)
terraform-lxd/terraform-provider-lxd: docs/resources/trust_token.md

> # lxd_trust_token

Reviewer: If you could clearly convey that those `lxd_trust_token` are meant to be issued by the LXD TF provider for the benefit of other clients wanting to interact with LXD rather than be used by the LXD TF provider itself.

> The `lxd_trust_token` resource allows you to create new tokens in the LXD trust store.

Reviewer: I'm not sure the tokens live in the trust store. They are a way to get your cert into the trust store though.

## Example Usage

```hcl
resource "lxd_trust_token" "token1" {
  name = "mytoken"
}

output "token" {
  value = lxd_trust_token.token1.token
}
```

## Argument Reference

* `name` - **Required** - Name of the token.

* `projects` - *Optional* - List of projects to restrict the token to.

* `remote` - *Optional* - The remote in which the resource will be created. If not provided,
  the provider's default remote will be used.

> * `trigger` - *Optional* - When to trigger the token generation. Possible values are `once` and `always` (if missing). Defaults to `always`.

Reviewer: Please have the default to `once`.

## Attribute Reference

The following attributes are exported:

* `token` - The generated token.

> * `expires_at` - Time when trust token expires. If token expiry is configured, the value will be in format `YYYY/MM/DD hh:mm TZ`.

Reviewer: ```suggestion
* `expires_at` - Time at which the trust token expires. If token expiry is configured, the value will be in format `YYYY/MM/DD hh:mm TZ`.
```

## Trust token expiry

~> **Warning**

>   The provider is unable to differentiate between the expired and consumed token.

Reviewer: ```suggestion
  The provider is unable to differentiate between an expired and a consumed token.
```

  If token generation is set to `once` and the token expires, it will not be regenerated.

> Trust token expiry is defined by the system configuration `core.remote_token_expiry`.

Reviewer: ```suggestion
Trust token expiry is defined server-side (`core.remote_token_expiry`).
```

If the setting is configured, `expires_at` attribute will be populated, otherwise, it will be empty.

For example, to set the token expiry to 20 minutes, run the following command:
```
lxc config set core.remote_token_expiry=20M
```

> If trigger is set to `once` the token will not be regenerated on consequite applies.

Reviewer: ```suggestion
If trigger is set to `once` the token will not be regenerated on subsequent plan applies.
```
